### ISSUE TITLE
Generate as property with initialization

### ISSUE THREAD

**Issue by RomanSoloweow** [AvaloniaUI/Avalonia.NameGenerator#28]
I think the generated types should be made properties.

For two reasons:
- property: 
 I use inheritance for my views. In the case of properties - I could mark the base class as abstract or use interfaces.

- Initialization: 
 With each call, the search for the element is repeated. I understand that Avalonia is fast in implementation, but even if the difference is in milliseconds, why do we need this delay if we can avoid it.

 ```C#
    internal Grid GridConnector { get; }
    internal TextBox TextBoxConnector { get;}
    internal Ellipse EllipseConnector { get; }
  
    public Connector()
    {
        GridConnector = this.FindControl<Grid>("GridConnector");
        TextBoxConnector = this.FindControl<TextBox>("TextBoxConnector");
        EllipseConnector = this.FindControl<Ellipse>("EllipseConnector");
    }
```

**Comment by worldbeater** [comment]
We can totally make the source generator behavior configurable, and allow switching the desired behavior based on an MSBuild property (the approach was introduced in https://github.com/AvaloniaUI/Avalonia.NameGenerator/pull/22!) So in the `ISourceGenerator.Execute` method we can obtain the option value via:
```cs
var generateInitializeComponentCalls = context
    .GetMSBuildProperty("AvaloniaNameGeneratorBehavior", "")
    .Equals("InitializeComponent", StringComparison.OrdinalIgnoreCase);
```

And enable a new `ICodeGenerator` implementation named e.g. `InitializeComponentCodeGenerator` (see https://github.com/AvaloniaUI/Avalonia.NameGenerator/blob/main/src/Avalonia.NameGenerator/Generator/FindControlCodeGenerator.cs for a reference implementation)

**Comment by kekekeks** [comment]
Do NOT generate constructors. Generate `void InitializeComponent()` instead

**Comment by worldbeater** [comment]
Yeah based on the conversation with @RomanSoloweow we are planning to try out the following:
```cs
public void InitializeComponent(bool loadXaml = true)
{
    if (loadXaml)
    {
        AvaloniaXamlLoader.Load(this);
    }

    GridConnector = this.FindControl<Grid>("GridConnector");
    TextBoxConnector = this.FindControl<TextBox>("TextBoxConnector");
    EllipseConnector = this.FindControl<Ellipse>("EllipseConnector");
}
```
If this approach works, we'll get an `InitializeComponent` API which is similar to WPF, WinForms, UWP etc.

**Comment by worldbeater** [comment]
This is now shipped as `0.3.1-preview` to NuGet.
